src/world.rs

```diff
@@ -55,13 +55,18 @@ pub struct World {
 impl World {
     /// Create an empty world
     pub fn new() -> Self {
+        // AtomicU64 is unsupported on 32-bit MIPS and PPC architectures
+        // For compatibility, use Mutex<u64>
         static ID: Mutex<u64> = Mutex::new(1);
 
+        let new_id = ID.lock().checked_add(1).unwrap();
+        *ID.lock() = new_id;
```

Comment:
`lock`ing twice introduces a race condition; if two worlds are constructed concurrently, both might read the same value from `ID`, increment it, and store the same value back, resulting in `ID` being incremented once and two worlds having the same `id` member. We must lock it once, hold that lock until the updated value is written, and then release it before performing any other work.

Reply:
Sorry, I don't understand how to do that. The only way to get access to the inner u64 that I see is to lock the mutex. So I don't know how to write to it without locking it again.

Reply:
`lock` returns a (smart) reference. Store that reference, then operate on it as many times as you like, then drop it when you're done.

Reply:
Oops, I thought I tried this, but forgot to writeback to ID.